src/containers/DataTable/index.jsx

```diff
@@ -0,0 +1,47 @@
+import React, { useState } from 'react';
+
+import Pagination from '../../components/Pagination';
+import Table from '../../components/Table';
+import Search from '../../components/Search';
+import {
+  calculateTotalNumberOfPages,
+  getSearchResult,
+  getCurrentPageData,
+} from './helpers';
+
+const DataTable = ({ rows, rowsPerPage = 40 }) => {
+  const [currentPageNumber, setCurrentPageNumber] = useState(0);
+  const [queryString, setQueryString] = useState('');
+
+  const filteredRows = getSearchResult(rows, queryString);
+  const totalNumberOfPages = calculateTotalNumberOfPages(
+    filteredRows.length,
+    rowsPerPage
+  );
+  const currentDisplayRows = getCurrentPageData(
+    filteredRows,
+    currentPageNumber,
+    rowsPerPage
+  );
+
+  const searchHandler = (text) => {
+    setQueryString(text);
+    setCurrentPageNumber(0);
+  };
+
+  const changeToPageNumber = (pageNumber) => setCurrentPageNumber(pageNumber);
+
+  return (
```

Comment:
After refactoring, the container only dispatches data to the children instead of needing to worry about how to render.